src/lib.rs

```diff
@@ -1454,6 +1556,16 @@ impl ServiceInfo {
         &self.addresses
     }
 
+    /// Insert an additional address to the service
+    pub fn insert_address(&mut self, addr: Ipv4Addr) {
```

Comment:
sure, will do

Reply:
Btw, since you defined `AsIpv4Addrs`,  can you use that trait for this method?  

As `Ipv4Addr` in this lib currently is from `Nix` crate, and I wanted to avoid exposing data structure from `Nix` directly to our caller.  Using a trait instead would be useful for the caller, similar like in `new()`.

Reply:
I thought it might have been a good addition when you want to manipulate the `ServiceInfo` after `::new()`, but before registering it, but now that I think about it, I don't think there's a use-case for this functionality without a new command for the daemon.